### PR TITLE
Add a timeout to downloading a dataset and stop waiting if the job fails

### PR DESCRIPTION
**Pull request by kurtwheeler**
My end to end tests in refinebio were never failing because this would never give up waiting for the dataset to become processed.

I've tested this by downloading a dataset.

### CODE REVIEW

File: pyrefinebio/high_level_functions.py [hunk 1]
```diff
@@ -120,12 +121,26 @@ def download_dataset(
...
     while not dataset.check():
+        if not dataset.is_processing and not dataset.is_processed:
+            print(
+                "Dataset failed to process. The system may be experiencing issues."
+                " Please try again later."
+            )
+            return None
+        elif datetime.now() - start_time > timedelta(minutes=15):
+            print(
+                "Dataset not processed after 15 minutes. The system may be experiencing issues."
```
Comment (wvauclain) [inline pyrefinebio/high_level_functions.py:134]: How long do most datasets take to smash? If someone provides a somewhat large dataset that takes more than 15 minutes to smash, so they would get a message that "The system may be experiencing issues. Please try again later.", which might cause them to think that the system is broken so they should resubmit the job.

**Review by wvauclain — APPROVED** [review]
Looks good to me, I have one question though.

File: pyrefinebio/high_level_functions.py (hunk 1 above)
Comment (kurtwheeler) [reply pyrefinebio/high_level_functions.py:134]: Hmm, this is actually a really good question. It looks like some of the smasher jobs have taken hours. Now I'm not sure what I want to do about this....

**Review by wvauclain — APPROVED** [review]
LGTM

**Comment by kurtwheeler** [comment]
I changed `http` to `api_interface` because it can cause conflicts with imports with the `http` library. Doing this required updating a bunch of files. Updating those files triggered my autoformatter, and now this PR is huge. Sorry!

**Pull request merged**